### PR TITLE
[Backend] Fix auth error in uat/prod for async jobs endpoint

### PR DESCRIPTION
**Pull request by sunil-sadasivan**
Connects: #3179 

This moves the jobs endpoint to the api/v1 route.  We needed to do this otherwise we'd hit authorization errors in UAT/PROD. 

To test:
```
curl localhost:3000/api/v1/jobs/start_async -d  'job_type=heartbeat' -H "Authorization: Token token=PUBLICDEMO123"
```

### CODE REVIEW

File: app/controllers/api/v1/jobs_controller.rb [hunk 1]
```diff
@@ -0,0 +1,22 @@
+class Api::V1::JobsController < Api::V1::ApplicationController
+  rescue_from NameError, TypeError, with: :unrecognized_job
+
+  def start_async
+    # available jobs supported by this endpoint
+    @jobs = {
```
Comment (joofsh) [inline app/controllers/api/v1/jobs_controller.rb]: Can we make this a constant outside the method?

File: config/routes.rb [hunk 2]
```diff
@@ -24,6 +24,7 @@
   namespace :api do
     namespace :v1 do
       resources :appeals, only: :index
+      post 'jobs/start_async' => 'jobs#start_async'
```
Comment (joofsh) [inline config/routes.rb]: Do you imagine a scenario where we'd want to have a POST request trigger the job synchronously? I sense we'd always want them to be async in which case the route could be more REST-y as `POST /jobs` alone. Then you could rename the method from `start_async` to `create`, which is more "Rails-y", but I'm not wed to that being a good thing, especially if you disagree and prefer the clarity here

File: app/controllers/api/v1/jobs_controller.rb [hunk 3]
```diff
@@ -0,0 +1,22 @@
+class Api::V1::JobsController < Api::V1::ApplicationController
+  rescue_from NameError, TypeError, with: :unrecognized_job
```
Comment (joofsh) [inline app/controllers/api/v1/jobs_controller.rb]: So this works with this route given that the code method is relatively simple and I _think_ there is only one point where NameError and TypeError could be thrown. But in general I'd rather us not rescue from these sorts of generic errors. It feels like we're losing control of the code a little bit, as these could be raised in other many other unintended scenarios. Looking at this code as a new developer I wouldn't necessarily have a full sense of where/when/how these might be raised.  Alternatively we could be more explicit on line ~14 with something like:

```
job = SUPPORTED_JOBS[params.require(:job_type)]
return unrecognized_job unless job
```

Comment (sunil-sadasivan) [reply app/controllers/api/v1/jobs_controller.rb]: Great point @joofsh, I agree with that!

File: app/controllers/api/v1/jobs_controller.rb (hunk 1 above)
Comment (sunil-sadasivan) [reply app/controllers/api/v1/jobs_controller.rb]: 👍 Yup that makes sense for the definition.

File: config/routes.rb (hunk 2 above)
Comment (sunil-sadasivan) [reply config/routes.rb]: Yeah I was going back and forth on clarity verses rails consistency, don't feel too strongly either way.  I _think_ there could be a possibility to start a job synchronously in the future but probably not something I can see we need today.  

We could easily do a `create` and pass in an option of say sync=1.  

Not feeling too strongly, feel like it's better to lean towards being consistent.

**Comment by sunil-sadasivan** [comment]
I've made an adjustment to use a more rails consistent api pattern.  Now the jobs endpoint should be triggered by:

```
curl localhost:3000/api/v1/jobs -d  'job_type=heartbeat' -H "Authorization: Token token=PUBLICDEMO123"
```

File: config/routes.rb [hunk 4]
```diff
@@ -24,6 +24,7 @@
   namespace :api do
     namespace :v1 do
       resources :appeals, only: :index
+      resources :jobs, only: :create
```
Comment (kierachell) [inline config/routes.rb]: I guess later it will be possible to have `:show` and use that for `MonitorBusinessCriticalJobsJob`

Comment (sunil-sadasivan) [reply config/routes.rb]: Good call @kierachell

**Pull request merged**